src/js/components/account/filters/objectClass/ObjectClassFilter.jsx

```diff
@@ -22,28 +22,33 @@ export default class ObjectClassFilter extends React.Component {
         this.toggleValue = this.toggleValue.bind(this);
     }
 
-    toggleValue(event) {
-        const code = event.target.value;
+    toggleValue(code) {
         this.props.updateFilter(code);
     }
 
     render() {
-        const items = Object.keys(objectClassDefinitions).map((code) => {
+        const items = Object.keys(objectClassDefinitions).map((code, index) => {
             const label = objectClassDefinitions[code];
-            const checked = this.props.selectedCodes.includes(code);
-            return (<ObjectClassItem
-                key={code}
-                code={code}
-                label={label}
-                checked={checked}
-                toggleValue={this.toggleValue} />);
+            return (<PrimaryCheckboxType
+                {...code}
+                {...this.props}
+                name={label}
+                value={code}
+                key={index}
```

Comment:
If `code` is unique (which I think it is, since the original implementation used it as the `key`), then it is [better to use](https://facebook.github.io/react/docs/reconciliation.html#keys) that as the `key` value rather than an index number. This is because the React can associate a specific data element to a component and watch both of those for changes. When an iterated number is used, React is more likely to re-render the component.